isy-security/src/test/java/de/bund/bva/isyfact/security/authentication/ResourceServerMultiTenancyTest.java

```diff
@@ -0,0 +1,121 @@
+    @RegisterExtension
+    public static final EmbeddedOidcProviderMock embeddedOidcProvider3 = new EmbeddedOidcProviderMock(HOST, PORT3, ISSUER_PATH3);
+    @LocalServerPort
+    private int port;
+
+    private String pingUri;
+
+    @BeforeEach
+    public void setup() {
+        embeddedOidcProvider1.removeAllClients();
+        embeddedOidcProvider2.removeAllClients();
+        embeddedOidcProvider3.removeAllClients();
+        pingUri = "http://localhost:" + port + "/ping";
+    }
+
+    /**
+     * Test with an issuer that is configured as a tenant in the resource server configuration.
+     */
+    @Test
+    public void shouldAllowPingFromTestrealm1() {
+        // manually get a token signed by the OIDC provider
+        String token = embeddedOidcProvider1.getAccessTokenString("cc-client", "testuser", Optional.empty(),
+                Collections.singleton("Rolle_A"));
+
+        String body = WebClient.create().get().uri(pingUri)
+                .header(HttpHeaders.AUTHORIZATION, "Bearer " + token)
+                .exchangeToMono(response -> {
+                    assertEquals(HttpStatus.OK, response.statusCode());
+                    return response.bodyToMono(String.class);
+                }).block();
+
+        assertEquals("true", body);
+
+    }
+
+    /**
+     * Test with an issuer that is configured as a tenant in the resource server configuration.
+     */
+    @Test
+    public void shouldAllowPingFromTestrealm2() {
+        // manually get a token signed by the OIDC provider
+        String token = embeddedOidcProvider2.getAccessTokenString("cc-client", "testuser", Optional.empty(),
+                Collections.singleton("Rolle_A"));
+
+        String body = WebClient.create().get().uri(pingUri)
+                .header(HttpHeaders.AUTHORIZATION, "Bearer " + token)
+                .exchangeToMono(response -> {
+                    assertEquals(HttpStatus.OK, response.statusCode());
+                    return response.bodyToMono(String.class);
+                }).block();
+
+        assertEquals("true", body);
+        verify(0, getRequestedFor(urlMatching(ISSUER_PATH2 + ".*")));
+    }
+
+    @Test
+    public void shouldDenyPingFromTestrealm3() {
+        // manually get a token signed by the OIDC provider
+        String token = embeddedOidcProvider3.getAccessTokenString("cc-client", "testuser", Optional.empty(),
+                Collections.singleton("Rolle_A"));
+
+        WebClient.create().get().uri(pingUri)
+                .header(HttpHeaders.AUTHORIZATION, "Bearer " + token)
+                .exchangeToMono(response -> {
+                    // status 401 Unauthorized expected as the tenant was not set
+                    assertEquals(HttpStatus.UNAUTHORIZED, response.statusCode());
+                    return response.bodyToMono(String.class);
+                }).block();
+    }
```

Comment:
Da aus dem Inhalt hier in dieser Datei nicht ersichtlich ist, dass für tenant-3 keine Konfigurations-Properties vorliegen, wäre es hilfreich per Kommentar zu dokumentieren, warum der Test mit Provider-3 abgelehnt wird. Denn Code-technisch sind alle drei Provider gleich aufgesetzt und es lässt sich erst mit Blick in die Konfiguration erahnen.

Reply:
Kommentare wurde entsprechend hinzugefügt.